src/goto-programs/graphml_witness.cpp

```diff
@@ -262,73 +438,85 @@ void graphml_witnesst::operator()(const goto_tracet &goto_trace)
     case goto_trace_stept::typet::ASSIGNMENT:
     case goto_trace_stept::typet::ASSERT:
     case goto_trace_stept::typet::GOTO:
+    case goto_trace_stept::typet::SPAWN:
+    {
+      xmlt edge(
+        "edge",
+        {{"source", graphml[from].node_name},
+         {"target", graphml[to].node_name}},
+        {});
+
       {
-        xmlt edge(
-          "edge",
-          {{"source", graphml[from].node_name},
-           {"target", graphml[to].node_name}},
-          {});
+        xmlt &data_f = edge.new_element("data");
+        data_f.set_attribute("key", "originfile");
+        data_f.data = id2string(graphml[from].file);
 
-        {
-          xmlt &data_f=edge.new_element("data");
-          data_f.set_attribute("key", "originfile");
-          data_f.data=id2string(graphml[from].file);
+        xmlt &data_l = edge.new_element("data");
+        data_l.set_attribute("key", "startline");
+        data_l.data = id2string(graphml[from].line);
 
-          xmlt &data_l=edge.new_element("data");
-          data_l.set_attribute("key", "startline");
-          data_l.data=id2string(graphml[from].line);
-        }
+        xmlt &data_t = edge.new_element("data");
+        data_t.set_attribute("key", "threadId");
+        data_t.data = std::to_string(it->thread_nr);
+      }
 
-        if(it->type==goto_trace_stept::typet::ASSIGNMENT &&
-           it->full_lhs_value.is_not_nil() &&
-           it->full_lhs.is_not_nil())
+      const auto lhs_object = it->get_lhs_object();
+      if(
+        it->type == goto_trace_stept::typet::ASSIGNMENT &&
+        lhs_object.has_value())
+      {
+        const std::string &lhs_id = id2string(lhs_object->get_identifier());
+        if(lhs_id.find("pthread_create::thread") != std::string::npos)
+        {
+          xmlt &data_t = edge.new_element("data");
+          data_t.set_attribute("key", "createThread");
+          data_t.data = std::to_string(++thread_id);
+        }
+        else if(
+          !contains_symbol_prefix(
```

Comment:
Comment on why dynamic objects are excluded?

Reply:
Comment is in the commit message: dynamic objects aren't part of the input source code, and therefore aren't known to any witness validator.